content/api/webhooks.apib

```diff
@@ -4,73 +4,80 @@ description: Manage webhooks, which enable you to receive push updates of the ra
-
-__**The SparkPost webhooks API uses MaxMind software [MaxMind License](https://www.maxmind.com/download/geoip/database/LICENSE.txt)**__
-
-## Events Documentation [/webhooks/events/documentation]
-
-### Documentation [GET]
-
-List descriptions of the events, event types, and event fields that could be included in a Webhooks post to your target URL.
-
-+ Request
-
-    + Headers
-
-            Accept: application/json
-
-
-## Events Samples [/webhooks/events/samples{?events}]
-
-### Samples [GET]
-
-List an example of the event data that will be posted by a Webhook for the specified events.
-
-<banner status="warning"><strong>Note</strong>: the data that will arrive at your target URL during normal operation (as opposed to calls to this endpoint) will <strong>not</strong> contain the top level <code>results</code> key shown in the example response.</banner>
-
-<banner status="info"><a href="https://www.sparkpost.com/enterprise-email/"><strong>SparkPost Enterprise</strong></a> customers may find the <code>delv_method</code> key useful to differentiate between Email, Push, and SMS type events.</banner>
 
-+ Parameters
-    + events (optional, string, `bounce`) ... Event types for which to get a sample payload, use the Webhooks Events endpoint to list the available event types, defaults to all event types.
-
-+ Request
+## Webhooks Object
+
++ Data Structure: Attributes
+    + name (string) - Name for webhook
+    + target (string) - URL of the target to which to POST event batches. 
+    + events (array)  - Array of event types this webhook will send. Use the [Events Documentation](#webhooks-documentation-get) endpoint to list the available event types.
+    + active (boolean) - The status of the webhook. When false, the target will no longer receive batches.
+    + custom_headers (object) - Object of custom headers to be used during POST requests to target.
+    + auth_type (enum) - Type of authentication to be used during POST requests to target. Read more about [webhook authentication](https://www.sparkpost.com/docs/tech-resources/webhook-authentication/#webhook-authentication-set-up).
+        + none
+        + basic
+        + oauth2
+    + auth_request_details (object) - Object containing details needed to request authorization token for OAuth 2.0. This is required when `auth_type` is oauth2.
+        + url (string, required) - The URL for the authorization server.
+        + body (object, required) - The body to send in the request to the authorization server. This likely should contain the client ID, client secret, and grant type. 
+    + `auth_credentials` (object) - Object containing credentials needed to make authorized POST requests for Basic Authentication. This is required when `auth_type` is basic.
```

Comment:
Something like "When auth type is this object is generated from `auth_request_details`."